stream-chat-android/src/main/java/com/getstream/sdk/chat/utils/CollectionsUtils.kt

```diff
@@ -0,0 +1,34 @@
+package com.getstream.sdk.chat.utils
+
+// Use when it is known that the element is always close to the end
+internal fun <T> List<T>.reverseFind(predicate: (T) -> Boolean): T? {
+    for (i in (this.size - 1) downTo 0) {
+        if (predicate(this[i])) {
+            return this[i]
+        }
+    }
+
+    return null
+}
+
+// Use when it is known that the element is always close to the end
+internal fun <T> List<T>.reverseIndexOf(element: T): Int {
+    for (i in (this.size - 1) downTo 0) {
+        if (this[i] == element) {
+            return i
+        }
+    }
+
+    return -1
+}
+
+// Use when it is known that the element is always close to the end
+internal fun <T> List<T>.reverseIndexOf(element: T, biPredicate: (T, T) -> Boolean): Int {
```

Comment:
Same here:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/index-of-last.html

Reply:
I this case I needed the `biPredicate`. Because I got some errors when simply looking for the element, the counter found messages that were much different than the true last one, the problem doesn't happens using getStableId... I can either use this function of override the equals from MessageListItem, what do you think?

Reply:
Nevermind, Martón presented a much better option.